packages/core/test/Form_test.js

```diff
@@ -375,7 +375,10 @@ describeRepeated("Form common", createFormComponent => {
   });
 
   describe("Custom submit buttons", () => {
-    it("should submit the form when clicked", done => {
+    // Skipping test due to regression in JSDOM  that causes it to not handle
+    //  form subsmissions from click events properly
+    // https://github.com/jsdom/jsdom/issues/3117
+    it.skip("should submit the form when clicked", done => {
```

Comment:
I spent some time trying to work around this regressions, but it wasn't easy. I chose to skip this test instead, in hope that it will be fixed soon. https://github.com/jsdom/jsdom/issues/3117

Reply:
Hmm, I'm not comfortable with disabling tests unless we really need to do so. Maybe we could try upgrading to a version of jsdom before the regression?

Reply:
Spent some time on this again and actually figured out the issue - apparently submit events are not fired from buttons if they are disconnected from the document tree. Mind reviewing again? :)

Reply:
Great!